timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/TestableTimelockServerConfiguration.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface TestableTimelockServerConfiguration {
+    TemplateVariables templateVariables();
+
+    @Value.Default
+    default boolean needsPostgresDatabase() {
+        return false;
+    }
+}
```

Comment:
I wonder if this the best we can do design wise, I feel like having these layers makes code a bit harder to read.

Reply:
What alternatives do you have in mind? The main ones I'm aware of are:
- passing a boolean and/or context directly into the testable timelock cluster (also OK with that, though it seems the property of whether you need a postgres is dependent on an individual server's config)
- having a template variable that is the yml-serialized form of the database bound persistence (don't like this; the freemarker templates don't natively support structured formats other than xml and html I think)

Reply:
I see, what you have now is fine.